src/appMain/life_cycle.cc

```diff
@@ -90,34 +90,44 @@ LifeCycle::LifeCycle()
 
 bool LifeCycle::StartComponents() {
   LOG4CXX_AUTO_TRACE(logger_);
-  transport_manager_ = transport_manager::TransportManagerDefault::instance();
-  DCHECK(transport_manager_ != NULL);
-
-  protocol_handler_ = new protocol_handler::ProtocolHandlerImpl(
-      transport_manager_,
-      profile::Profile::instance()->message_frequency_time(),
-      profile::Profile::instance()->message_frequency_count(),
-      profile::Profile::instance()->malformed_message_filtering(),
-      profile::Profile::instance()->malformed_frequency_time(),
-      profile::Profile::instance()->malformed_frequency_count(),
-      profile::Profile::instance()->multiframe_waiting_timeout());
-  DCHECK(protocol_handler_ != NULL);
-
-  connection_handler_ = connection_handler::ConnectionHandlerImpl::instance();
-  DCHECK(connection_handler_ != NULL);
 
+  DCHECK(!transport_manager_);
+  transport_manager_ = transport_manager::TransportManagerDefault::instance();
+  DCHECK(transport_manager_);
+
+  DCHECK(!connection_handler_);
+  connection_handler_ = new connection_handler::ConnectionHandlerImpl(
+                          *profile::Profile::instance(),
+                          *transport_manager_);
+  DCHECK(connection_handler_);
+
+  DCHECK(!protocol_handler_);
+  //TODO(AKutsan) : APPLINK-20265 Singletons should be removed
```

Comment:
Are you sure that TODO comment with Jira Task ID APPLINK-.... must present in open source repo?